src/main/java/core/basesyntax/ArrayList.java

```diff
@@ -1,48 +1,115 @@
 package core.basesyntax;
 
+import java.util.NoSuchElementException;
+import java.util.Objects;
+
 public class ArrayList<T> implements List<T> {
+    private static final int DEFAULT_CAPACITY = 10;
+    private int size;
+    private T[] array;
+
+    public ArrayList(int capacity) {
+        if (capacity <= 0) {
+            throw new IllegalArgumentException("Array`s capacity cannot be <= 0");
+        } else {
+            array = (T[]) new Object[capacity];
+            this.size = 0;
+        }
+    }
+
+    public ArrayList() {
+        array = (T[]) new Object[DEFAULT_CAPACITY];
+        this.size = 0;
```

Comment:
In Java, declaring the initial value of a variable when it's first declared is optional. When you write private int size = 0;, you're explicitly setting the initial value of size to 0. However, in Java, if you don't initialize an instance variable, it gets a default value based on its type. For int, the default initial value is 0 anyway. So, writing private int size = 0; is redundant in this case because size would already be initialized to 0 by default. You could simply write private int size; to achieve the same result.